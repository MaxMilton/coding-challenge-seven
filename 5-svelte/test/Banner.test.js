'use strict';

const Banner = require('../src/Banner.html');

describe('Banner component', () => {
  it.skip('throws error with no props', () => {
    expect.assertions(1);
    function wrapper() {
      const target = document.createElement('div');
      new Banner({ target });
    }
    expect(wrapper).toThrow();
  });

  it('renders correctly', () => {
    expect.assertions(3);
    const target = document.createElement('div');
    new Banner({
      target,
      data: {
        tile: '../tile.jpg',
        title: 'Home And Away',
      },
    });
    expect(target.querySelector('.banner > img.tile')).not.toBeNull();
    expect(target.querySelector('.overlay > h3.title')).not.toBeNull();
    expect(target.innerHTML).toMatchSnapshot();
  });
});
