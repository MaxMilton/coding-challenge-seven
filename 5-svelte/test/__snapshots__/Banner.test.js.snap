// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component renders correctly 1`] = `
<div class="banner">
  <img src="../tile.jpg" alt="Home And Away" class="tile">

  <div class="overlay">
    <img src="../logo.png" alt="7plus" class="network">

    <h3 class="title">Home And Away</h3>
  </div>
</div>
`;
